src/services/codefixes/convertToAsyncFunction.ts

```diff
@@ -0,0 +1,455 @@
+        }
+
+        forEachChild(nodeToRename, function visit(node: Node) {
+
+            if (isIdentifier(node)) {
+                const type = checker.getTypeAtLocation(node);
+                const symbol = checker.getSymbolAtLocation(node);
+                const newName = getNewNameIfConflict(node, allVarNames);
+
+                if (symbol && !node.symbol) {
+                    node.symbol = symbol;
+                }
+
+                // if the identifier refers to a function
+                if (symbol && type && type.getCallSignatures().length > 0 && isFunctionRef(node) && definedInFile(symbol, context.sourceFile)) {
+                    if (type.getCallSignatures()[0].parameters.length) {
+                        // add the new synthesized variable for the declaration (ex. blob in let blob = res(arg))
+                        const synthName = createIdentifier(type.getCallSignatures()[0].parameters[0].name);
+                        allVarNames.push(synthName);
+                        synthNamesMap.set(String(getSymbolId(symbol)), [synthName, allVarNames.filter(elem => elem.text === synthName.text).length]);
+                    }
+                }
+                else if (symbol && definedInFile(symbol, context.sourceFile)) {
+                    let setName = false;
+
+                    for (const ident of allVarNames) {
+                        if (ident.text === node.text && ident.symbol !== node.symbol) {
+                            allVarNames.push(newName[0]);
+                            synthNamesMap.set(String(getSymbolId(symbol)), newName);
+                            setName = true;
+                        }
+                    }
+
+                    if (!setName) {
+                        allVarNames.push(node);
+                        synthNamesMap.set(String(getSymbolId(symbol)), [getSynthesizedDeepClone(node), allVarNames.filter(elem => elem.text === node.text).length]);
+                    }
+                }
+            }
+            else {
+                forEachChild(node, visit);
+            }
+        });
+
+        return getSynthesizedDeepClone(nodeToRename, /*includeTrivia*/ true, synthNamesMap, checker);
+    }
+
+    function getNewNameIfConflict(name: Identifier, allVarNames: Identifier[]): [Identifier, number] {
+        const numVarsSameName = allVarNames.filter(elem => elem.text === name.text).length;
+        return numVarsSameName === 0 ? [name, 1] : [createIdentifier(name.text + "_" + numVarsSameName), numVarsSameName];
+    }
+
+    function returnsAPromise(node: Expression, checker: TypeChecker): boolean {
+        const nodeType = checker.getTypeAtLocation(node);
+        if (!nodeType) {
+            return false;
+        }
+
+        return !!checker.getPromisedTypeOfPromise(nodeType) && (!isCallExpression(node) || !isCallback(node, "then", checker) && !isCallback(node, "catch", checker) && !isCallback(node, "finally", checker));

```

Comment:
move the syntactic checks earlier, these are cheaper to do. the checks involving the type are more expensive, so leave these to the end.

similarly, i would not even call `getTypeAtLocation` if the node is not a callExpression